build/worker/Dockerfile

```diff
@@ -0,0 +1,116 @@
+# syntax=docker/dockerfile:1.3
+
+ARG PYTHON_VERSION
+FROM python:${PYTHON_VERSION}-slim-bullseye as base
+
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+    # Postgres
+    libpq-dev \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN useradd -ms /bin/bash frappe
+USER frappe
+RUN mkdir -p /home/frappe/frappe-bench/apps /home/frappe/frappe-bench/logs /home/frappe/frappe-bench/sites
+WORKDIR /home/frappe/frappe-bench
+
+RUN pip install --no-cache-dir -U pip wheel \
+    && python -m venv env \
+    && env/bin/pip install --no-cache-dir -U pip wheel
+USER root
+
+
+
+FROM base as build_deps
+
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+    # Install git here because it is not required in production
+    git \
+    # gcc and g++ are required for building different packages across different versions
+    # of Frappe and ERPNext and also on different platforms (for example, linux/arm64).
+    # It is safe to install build deps even if they are not required
+    # because they won't be included in final images.
+    gcc \
+    g++ \
+    && rm -rf /var/lib/apt/lists/*
+USER frappe
+
+
+
+FROM build_deps as frappe_builder
+
+ARG FRAPPE_VERSION
+RUN git clone --depth 1 -b ${FRAPPE_VERSION} https://github.com/frappe/frappe apps/frappe \
+    && env/bin/pip install --no-cache-dir -e apps/frappe \
+    && rm -r apps/frappe/.git
```

Comment:
Sometimes it's helpful to identify or verify the latest commit hash and know if the container is running the required code.

Reply:
It is a good practice to remove `.git` folder (usually it is done in `.dockerignore` in case if Dockerfile located in main repo. If something changes in code, version of an app should change, too.

Reply:
ideally yes version should change.

Frappe bump doesn't cause change in ERPNext version